### DISCUSSION TITLE
az ad sp create-for-rbac 에러

### DISCUSSION THREAD

**Discussion by DoItAmelia** [devrel-kr/HackaLearn#116] [2 Q&A - 클라우드 스킬 챌린지] [answered]
안녕하세요!
github-actions-continuous-delivery-azure >
Create a staging workflow #1 >
Step 3: Set up the environment for your app >
중 아래 과정을 시행하던 중에
![image](https://user-images.githubusercontent.com/88205708/128654155-a4234b0b-84dc-48de-b2d6-b97aed303111.png)
 
Insufficient privileges to complete the operation. 라는 문구와 함께 다음 단계로 넘어가지 않습니다.
확인 결과 해당 구독에 대한 제 역할은 '소유자'입니다. 
![image](https://user-images.githubusercontent.com/88205708/128654236-bfe84e31-a9b8-473f-b2f8-89ae6d705d65.png)
![image](https://user-images.githubusercontent.com/88205708/128654257-9f3eaa8d-4f30-403a-bcb0-a10a226cba9c.png)
![image](https://user-images.githubusercontent.com/88205708/128654261-00bcd22c-143f-48aa-9b26-df4de48e66af.png)

**Comment by ingHHK** [comment]
[https://docs.microsoft.com/ko-kr/azure/role-based-access-control/built-in-roles](url)에 따르면 소유자는 'Azure RBAC에서 역할을 할당하는 기능을 포함하여 모든 리소스를 관리할 수 있는 모든 권한을 부여합니다.'로 역할에는 문제가 없어 보입니다.

**Reply by DoItAmelia** [reply]
감사합니다. 
제가 찾아본 바로 해당 에러는 역할과 관련된 문제인데, 역할에는 아무런 문제가 없었기에
1. 컴퓨터를 껐다 켜본다
2. 새로운 구독을 만들어본다
이와같은 방법들도 다 해보았는데 여전히 동일한 에러가 발생합니다. 
혹시 또 추가로 시도해볼 방법이 있을까요?

**Reply by ingHHK** [reply]
윈도우 환경에서 실습 중이시라면 해당 명령어를 메모장에 복사 붙여넣기 한 후 공백을 제거해 시도해보세요!

az ad sp create-for-rbac --name "GitHub-Actions" --role contributor --scopes /subscriptions/{subscription-id} --sdk-auth

직접 타이핑하면 오타가 날 수 있습니다.
az login 다시 한 다음에 위 명령어에서 구독 id만 수정해서 복붙해 실행해보세요!

**Reply by DoItAmelia** [reply]
말씀해주신대로 위 명령어에서 구독 id만 수정한 후 그대로 복붙해 실행해보았는데 여전히 같은 에러가 발생합니다. ㅠㅠ
![image](https://user-images.githubusercontent.com/88205708/128661326-5480b321-ede6-4fd8-aa21-aeb8721ef9eb.png)

**Reply by ingHHK** [reply]
![image](https://user-images.githubusercontent.com/30424381/128661874-6bd3814e-92bb-4391-bd8c-22e9f6670b28.png)

저는 정상적으로 됩니다만,,, 흠,,, 다음 단계로 'Creating 'contributor' role assignment under scope'가 진행돼야 하는데 안되는 것이군요,,,

**Reply by ingHHK** [reply]
계정이 학교 등 기관에서 제공하는 계정인가요? 아니면 개인 계정인가요? 계정에 따라 접근할 수 있는 자원이 제한되나봐요.
Step 3에 보면 'If your Azure account is created through work, you may encounter issues accessing the necessary resources -- we recommend creating a new account for personal use and for this course.'라고 되어 있네요.

**Reply by DoItAmelia** [reply]
저는 학교 기관에서 제공하는 계정을 사용하였습니다!
조원 중 한명은 저와 같은 학교 계정을 사용하고, 한명은 개인 계정을 사용하였는데, 둘 다 오류없이 pass하였습니다.

**(임시) 해결
일단 제 다른 microsoft계정을 사용하여 azure 계정을 새롭게 만든 뒤, 구독을 만들어서 해봤더니 에러 없이 해결되었습니다! 
우선 이 계정으로 계속 진행해보겠습니다. 감사합니다:)
![image](https://user-images.githubusercontent.com/88205708/128662711-7ea879bb-ad79-4026-9b6c-925bca40cdf8.png)

**Reply by ingHHK** [reply]
그렇군요.. 해당 계정에만 이슈가 있나보네요... 알겠습니다~!

**Comment by justinyoo** [comment, marked as answer]
1. 가장 먼저 `az account list` 명령어를 통해 현재 로그인 한 이메일 계정에 연결된 구독 갯수를 확인해 보세요. 구독 갯수가 하나라면 상관 없습니다만, 여러개라면 확인할 것이 몇가지 더 있습니다.
2. 그리고, `az accoutn show` 명령어를 통해 현재 활성화된 구독을 확인해 보세요. 만약 현재 활성화되어 있는 구독이 내가 사용하려고 하는 구독과 같으면 상관 없습니다.
3. 만약 다르다면 `az account set --subscription <subscriptionID>` 명령어를 통해 내가 사용하고자 하는 구독을 활성화해 주세요.

이렇게 확인한 후 위의 명령어 `az ad sp create-for-rbac ...`을 실행시켰는데도 동일하게 권한 문제가 나온다면, 현재 로그인한 계정이 해당 구독에 대해 저 명령어를 실행시킬 수 있는 권한이 없는 상태입니다. 이 때는 학교 관리자에게 문의하셔서 권한을 받아 오시거나, 지금 하신 대로 일단 별도의 무료 계정을 하나 생성하셔서 사용하시면 됩니다.
